src/snowflake/snowpark/_internal/analyzer/table_function.py

```diff
@@ -88,7 +88,7 @@ def __init__(
         self.path = path
         self.outer = outer
         self.recursive = recursive
-        self.mode = mode
+        self.mode = mode.upper()
```

Comment:
mode should be one of "OBJECT", "ARRAY", "BOTH" and typically the caller has been upper-ing this.

instead of changing this to mode.upper() you might change the signature to Literal["OBJECT", "ARRAY", "BOTH"]

Reply:
this is a refactor we did approved by the snowpark team. I agree that this would be the correct way, yet sometimes literals like `object` or so are used. By adding the logic here to store a well-formed string, we were able to simplify logic in the corresponding functions. (Before it was something like `if self.mode.upper() == "OBJECT" ...`)